### ISSUE TITLE
Resolve Vertex Scheduler Character Limit Issue for Long Pipeline Names

### ISSUE THREAD

**Issue by strickvl** [zenml-io/zenml#2262]
## Open Source Contributors Welcomed!
Please comment below if you would like to work on this issue!

### Contact Details [Optional]

support@zenml.io

## What happened?
A user encountered an issue where the Vertex scheduler in ZenML does not accept pipeline names longer than 64 characters. This limitation requires users to shorten their pipeline names to successfully schedule their pipelines on Vertex AI, which can be restrictive and inconvenient.

## Task Description
Investigate and address the issue with the Vertex scheduler's character limit for pipeline names in ZenML. The goal is to either increase the character limit or implement a mechanism that automatically handles longer names without user intervention.

## Expected Outcome
Users should be able to schedule pipelines on Vertex AI without the constraint of a 64-character limit on pipeline names.
ZenML should provide a seamless experience, either by accommodating longer names or by intelligently managing them within the existing Vertex scheduler limitations.
This fix will enhance user convenience and flexibility in naming pipelines.

## Steps to Implement
- Reproduce the issue to understand the exact limitations and behavior of the Vertex scheduler with respect to pipeline names.
- Explore solutions such as increasing the character limit or automatically abbreviating/transforming long pipeline names while preserving their uniqueness and recognizability.
- Implement the chosen solution in a way that is transparent and intuitive for the user.
- Test the implementation with various pipeline name lengths to ensure compatibility and functionality.
- Update documentation to inform users about the handling of pipeline names in Vertex scheduler and any best practices or limitations.

## Additional Context
This issue is crucial for users who require descriptive and potentially lengthy names for their pipelines, especially in complex or large-scale MLOps environments.

## Code of Conduct
- [ ] I agree to follow this project's Code of Conduct

**Comment by strickvl** [comment]
@christianversloot if I'm not completely mistaken, we had a conversation about this in Slack? Unfortunately the slack history hasn't been retained.

**Comment by christianversloot** [comment]
Correct, but that was for SageMaker orchestrator. There, I believe we limit name length to circumvent this error. It could perhaps be inspiration for a fix here :)

**Comment by strickvl** [comment]
@christianversloot ah yes! https://github.com/zenml-io/zenml/pull/1505 was the PR.

**Comment by ashutosh887** [comment]
@strickvl @christianversloot 
I would like to work on this... Please assign!

**Comment by Merthoshan** [comment]
has the bug been solved? if not i would like to solve it

**Comment by strickvl** [comment]
The bug has not been solved yet @Merthoshan. @ashutosh887 are you working on it, if not, can let @Merthoshan take a shot.

**Comment by ashutosh887** [comment]
Where can we discuss if we're facing some issues understanding the bug? @strickvl 
I've worked it

**Comment by strickvl** [comment]
Here is the place.

**Comment by Prabhat-Thapa45** [comment]
is there a simpler way to replicate this I could take this but It's going to be difficult for me to replicate this if this needs me to setup GCP which requires some cost. @strickvl

**Comment by RoblaR** [comment]
Hi! I’d like to work on this issue. My plan is to explore where the pipeline name is constructed and add a fix that truncates or sanitizes names to comply with Vertex’s 64-character limit. Could you please assign this to me? Thanks! @strickvl

**Comment by htahir1** [comment]
All the best @RoblaR !

**Comment by konsougiou** [comment]
Hey! Could I also be assigned to this issue?

**Comment by konsougiou** [comment]
Hey @strickvl , has this issue been resolved? I have not been able to replicate the bug?
